modules/light-clients/08-wasm/types/store.go

```diff
@@ -123,3 +129,46 @@ func (s storeAdapter) Iterator(start, end []byte) wasmvmtypes.Iterator {
 func (s storeAdapter) ReverseIterator(start, end []byte) wasmvmtypes.Iterator {
 	return s.parent.ReverseIterator(start, end)
 }
+
+func getClientID(clientStore storetypes.KVStore) (string, error) {
+	upws, isUpdateProposalWrappedStore := clientStore.(updateProposalWrappedStore)
+	if isUpdateProposalWrappedStore {
+		// if the clientStore is a updateProposalWrappedStore, we retrieve the subjectStore
+		// because the contract call will be made on the client with the ID of the subjectStore
+		clientStore = upws.subjectStore
+	}
```

Comment:
I can see a case for moving this logic to wasmCall after checking payload is `CheckSubstituteAndUpdateState`. However, since CheckSubstituteAndUpdateState is the only user of this storage type, I don't see anything wrong with leaving it here.

Reply:
I agree